core/src/send_transaction_service.rs

```diff
@@ -208,49 +217,60 @@ impl SendTransactionService {
                 info!("Transaction is rooted: {}", signature);
                 result.rooted += 1;
                 inc_new_counter_info!("send_transaction_service-rooted", 1);
-                false
+                return false;
+            }
+            if let Some((nonce_pubkey, durable_nonce)) = transaction_info.durable_nonce_info {
+                let nonce_account = working_bank.get_account(&nonce_pubkey).unwrap_or_default();
+                if !nonce_account::verify_nonce_account(&nonce_account, &durable_nonce)
+                    && working_bank.get_signature_status_slot(signature).is_none()
```

Comment:
Should we still bail on nonced TXs after some time?  Seems like one could cause trouble by broadcasting a bunch and backing up the queue

Reply:
My initial thought is that's not much different than broadcasting a bunch of identical recent-blockhash transactions and backing up the queue. But I'm not opposed to some kind of timeout failsafe. I think that would be done by arbitrarily picking a `last_valid_slot` for durable-nonced txs (and then undoing the else-if set up in this PR).
Is 300 slots a reasonable timeout?

Reply:
> My initial thought is that's not much different than broadcasting a bunch of identical recent-blockhash transactions and backing up the queue.

My thinking is that the TXs could be maliciously constructed to conflict with one another, resulting in a financially bound total pool size, draining at one TX per block.  Where as recent-blockhash TXs would be limited to a pool size of `MAX_RECENT_BLOCKHASHES` or else it starts draining at `N` TX per block where `N = POOL_SIZE / MAX_RECENT_BLOCKHASHES`

> But I'm not opposed to some kind of timeout failsafe. I think that would be done by arbitrarily picking a last_valid_slot for durable-nonced txs (and then undoing the else-if set up in this PR).

Looks like that would do the trick!

> Is 300 slots a reasonable timeout?

300 seems like as good a starting point as any!

Reply:
Implemented here: https://github.com/solana-labs/solana/pull/14308/commits/3470fb820bc059f2a24a9a24ce40b2c2eb444970